R/install_torsten.R

```diff
@@ -0,0 +1,197 @@
+              "\nrelease_url will be ignored.", call. = FALSE)
+    }
+
+    release_list <- get_torsten_release_list()
+    release <- release_list[grep(version, release_list)]
+    if(length(release) < 1){
+      stop("Available Torsten versions do not include ", version, call. = FALSE)
+    }
+    download_url <- paste0("https://github.com/metrumresearchgroup/Torsten/archive/refs/tags/",
+                          release, ".tar.gz")
+  }
+  if (!is.null(release_url)) {
+    if (!endsWith(release_url, ".tar.gz")) {
+      stop(release_url, " is not a .tar.gz archive!",
+           "cmdstanr supports installing from .tar.gz archives only.", call. = FALSE)
+    }
+    download_url <- release_url
+    message("* Installing Torsten from ", release_url)
+  } else {
+    if(is.null(version)) {
+      ver <- latest_torsten_release()
+      message("* Latest Torsten release is ", ver)
+      download_url <- paste0("https://github.com/metrumresearchgroup/Torsten/archive/refs/tags/",
+                            ver, ".tar.gz")
+    }
+  }
+  message("* Installing Torsten from ", download_url)
+  dir_torsten <- file.path(dir, substr(basename(download_url), 1, nchar(basename(download_url)) - 7))
+  dir.create(dir_torsten, recursive = TRUE, showWarnings = FALSE)
+  dest_file <- file.path(dir_torsten, basename(download_url))
+  dir_cmdstan <- file.path(dir_torsten, "cmdstan")
+  ## Reset timeout for download. The 60 s default is not enough.
+  options(timeout = max(300, getOption("timeout")))
+  tar_downloaded <- try(suppressWarnings(utils::download.file(url = download_url,
+                                                              destfile = dest_file, quiet = quiet)),
+                        silent = TRUE)
+  if (tar_downloaded != 0) {
+    if (!is.null(version)) {
+      stop("Download of Torsten failed. Please check if the supplied version number is valid.",
+           call. = FALSE)
+    }
+    if (!is.null(release_url)) {
+      stop("Download of Torsten failed. Please check if the supplied release URL is valid.",
+           call. = FALSE)
+    }
+    stop("Download of Torsten failed. Please try again.", call. = FALSE)
+  }
+  message("* Download complete")
+
+  message("* Unpacking archive...")
+  untar_rc <- utils::untar(
+    dest_file,
+    exdir = dir_torsten,
+    extras = "--strip-components 1"
+  )
+  if (untar_rc != 0) {
+    stop("Problem extracting tarball. Exited with return code: ", untar_rc, call. = FALSE)
+  }
+  file.remove(dest_file)
+  cmdstanr::cmdstan_make_local(dir = dir_cmdstan, cpp_options = cpp_options, append = TRUE)
+  # # Setting up native M1 compilation of CmdStan and its downstream libraries
+  # if (cmdstanr:::is_rosetta2()) {
+  #   cmdstanr::cmdstan_make_local(
+  #     dir = dir_cmdstan,
+  #     cpp_options = list(
+  #       CXX = "arch -arch arm64e clang++"
+  #     ),
+  #     append = TRUE
+  #   )
+  # }
+
+  message("* Building CmdStan binaries...")
```

Comment:
Although the current message isn't inaccurate (given Torsten builds on top of CmdStan), should this say "Torsten" instead?

Reply:
Changed to "Torsten".